### PR TITLE
Properly forward HTTP requests which are expected to have body

### PR DESCRIPTION
**Pull request by ichernetsky**
JIRA issues: MARATHON-8084

### CODE REVIEW

**Comment by timcharper** [comment]
@ichernetsky would it be possible to add an integration test for this?

File: src/main/scala/mesosphere/marathon/api/LeaderProxyFilter.scala [hunk 1]
```diff
@@ -216,11 +216,18 @@ class JavaUrlConnectionRequestForwarder(
         case "GET" | "HEAD" | "DELETE" =>
           leaderConnection.setDoOutput(false)
         case _ =>
+          // If Content-Type header is not set, and true is passed to setDoOutput,
+          // the header will be set to application/x-www-form-urlencoded.
```
Comment (alenkacz) [inline src/main/scala/mesosphere/marathon/api/LeaderProxyFilter.scala]: this is some kind of default of the underlying library we use?

Comment (ichernetsky) [reply src/main/scala/mesosphere/marathon/api/LeaderProxyFilter.scala]: Yes.

**Comment by ichernetsky** [comment]
Added an integration test. Please check it out.

**Comment by kensipe** [comment]
@ichernetsky could you rebase... from there it looks ready to merge

**Comment by kensipe** [comment]
@ichernetsky I did the rebase... and fixed some formatting... looking to merge after the test succeeds

**Pull request merged**